### ISSUE TITLE
[Feature Request] Timeline visibility of Boosts

### ISSUE THREAD

**Issue by BDGreyson** [tuskyapp/Tusky#3095]
The Mastodon web interface allows you to separately set the timeline visibility of a Boost when Boosting (visible, unlisted, followers-only).

This SEEMS to be lacking in Tusky, with the confirmation dialog only displaying one button.

It would be a nice addition to bring Tusky more in line with the web features.

Thank you either way!

* * * *
- Tusky Version: 20.0
- Android Version: Android 10
- Android Device: Huawei P30 Pro

- [X] I searched or browsed the repo’s other issues to ensure this is not a duplicate.

**Comment by connyduck** [comment]
I have never seen this feature and I can't find it on mastodon.social, can you please show me what exactly you mean?

**Comment by BDGreyson** [comment]
At least on my instance, and I assumed it's a general MAstodon feature, but feel free to correct me, when you boost a Toot, you get this attached window:
![grafik](https://user-images.githubusercontent.com/50461895/210044615-2ecf189e-61bd-40c1-82cf-41d86464108d.png)

where you can choose who to send the boost to:

![grafik](https://user-images.githubusercontent.com/50461895/210044675-d2a1f968-30a3-42e3-9ce2-71a8c246a848.png)

So, yeah, my instance has this, no clue if that is a specific configuration then.

Instance is social.tchncs.de

**Comment by connyduck** [comment]
Ah, it is only there when the confirmation dialog is on, it is actually a vanilla Mastodon feature. Yes we should support that.

**Comment by Exagone313** [comment]
If you enable the option `Show confirmation before boosting`, it shows a menu with the single `Boost` entry.

I suggest to replace this by three entries for the visibility of the boost (wording to choose):
* Public boost
* Unlisted boost
* Followers-Only boost

I'm willing to look into contributing for this.

**Comment by connyduck** [comment]
> I'm willing to look into contributing for this.

Yes please @Exagone313! Multiple options in the dialog sounds good. 
The [boost endpoint of the Mastodon Api](https://docs.joinmastodon.org/methods/statuses/#boost) has a `visibility` parameter, looks like setting that to the correct value should be all there is to do.
